gno.land/pkg/integration/testing_integration.go

```diff
@@ -369,6 +422,118 @@ func createAccount(env envSetter, kb keys.Keybase, accountName string) (gnoland.
 
 	return gnoland.Balance{
 		Address: address,
-		Amount:  std.Coins{std.NewCoin("ugnot", 1000000000000000000)},
+		Amount:  std.Coins{std.NewCoin("ugnot", 10e6)},
 	}, nil
 }
+
+type pkgsLoader map[string]gnomod.Pkg
+
+func (pkgs pkgsLoader) list() gnomod.PkgList {
+	list := make([]gnomod.Pkg, 0, len(pkgs))
+	for _, pkg := range pkgs {
+		list = append(list, pkg)
+	}
+	return list
+}
+
+func (pkgs pkgsLoader) Txs(creator bft.Address, fee std.Fee, deposit std.Coins) ([]std.Tx, error) {
+	pkgslist, err := pkgs.list().Sort()
+	if err != nil {
+		return nil, fmt.Errorf("uanble to sort packages: %w", err)
+	}
+
+	txs := make([]std.Tx, len(pkgslist))
+	for i, pkg := range pkgslist {
+		tx, err := gnoland.LoadPackage(pkg, creator, fee, deposit)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load pkg %q: %w", pkg.Name, err)
+		}
+		txs[i] = tx
+	}
+
+	return txs, nil
+}
+
+func (pkgs pkgsLoader) loadPackagesFromDir(path string) error {
+	// list all packages from target path
+	pkgslist, err := gnomod.ListPkgs(path)
+	if err != nil {
+		return fmt.Errorf("listing gno packages: %w", err)
+	}
+
+	for _, pkg := range pkgslist {
+		if _, ok := pkgs[pkg.Name]; !ok {
+			pkgs[pkg.Name] = pkg
+		}
+	}
+
+	return nil
+}
+
+func (pkgs pkgsLoader) loadPackages(modroot string, path, name string) error {
+	if path == "" {
+		return fmt.Errorf("no path specified for package")
+	}
+
+	pkg := gnomod.Pkg{
+		Dir:  path,
+		Name: name,
+	}
+
+	if pkg.Name == "" {
+		// try to load `gno.mod` informations
+		gnoModPath := filepath.Join(pkg.Dir, "gno.mod")
+		data, err := os.ReadFile(gnoModPath)
+		if os.IsNotExist(err) {
+			return fmt.Errorf("no name specified /gnomod doesn't exist (%q)", gnoModPath)
+		}
+
+		if err != nil {
+			return fmt.Errorf("unable to load gno.mod: %w", err)
+		}
+
+		gnoMod, err := gnomod.Parse(gnoModPath, data)
+		if err != nil {
+			return fmt.Errorf("%q parse gnomod error: %w", gnoModPath, err)
+		}
+
+		gnoMod.Sanitize()
+		if err := gnoMod.Validate(); err != nil {
+			return fmt.Errorf("validate %q error: %w", gnoModPath, err)
+		}
+
+		// override pkg info with mod infos
+		pkg.Name = gnoMod.Module.Mod.Path
+		pkg.Draft = gnoMod.Draft
+		for _, req := range gnoMod.Require {
```

Comment:
maybe we can use `gnomod.ParseGnoMod()`?

```suggestion
		gm, err := gnomod.ParseGnoMod(gnoModPath)
		if err != nil {
			return err
		}
		gm.Sanitize()

		// override pkg info with mod infos
		pkg.Name = gm.Module.Mod.Path
		pkg.Draft = gm.Draft
		for _, req := range gm.Require {
```